crates/signal/src/lib.rs

```diff
@@ -60,20 +58,164 @@
 //!
 //! `SPDX-License-Identifier: MIT OR Apache-2.0`
 #![cfg_attr(all(coverage_nightly, test), feature(coverage_attribute))]
-#![cfg(unix)]
 
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
-use tokio::signal::unix::{Signal, SignalKind};
+#[cfg(unix)]
+use tokio::signal::unix;
+#[cfg(unix)]
+pub use tokio::signal::unix::SignalKind as UnixSignalKind;
+#[cfg(windows)]
+use tokio::signal::windows;
 
 #[cfg(feature = "bootstrap")]
 mod bootstrap;
 
 #[cfg(feature = "bootstrap")]
 pub use bootstrap::{SignalConfig, SignalSvc};
 
-/// A handler for listening to multiple Unix signals, and providing a future for
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SignalKind {
+    Interrupt,
+    Terminate,
+    #[cfg(windows)]
+    Windows(WindowsSignalKind),
+    #[cfg(unix)]
+    Unix(UnixSignalKind),
+}
+
+#[cfg(unix)]
+impl From<UnixSignalKind> for SignalKind {
+    fn from(value: UnixSignalKind) -> Self {
+        Self::Unix(value)
```

Comment:
I think on this conversion we check if its an interrupt or terminate signal and if it is use the predefined type.